src/near-rpc-provider.ts

```diff
@@ -312,4 +300,27 @@ export class NearRpcProvider extends JsonRpcProvider {
       })
     }
   }
+
+  async getBlockWithChunk(params: Record<string, any>): Promise<GetBlockWithChunkRpcResponse> {
+    const getBlockChunkParams: GetBlockChunkParams = {}
+    try {
+      if (params.chunk_id) {
+        getBlockChunkParams.chunk_id = params.chunk_id
+      }
+
+      if (params.block_id && params.shard_id) {
+        getBlockChunkParams.block_id = params.block_id
+        getBlockChunkParams.shard_id = params.shard_id
+      }
+
+      const chunkResponse = await this.send<GetBlockWithChunkRpcResponse>('chunk', getBlockChunkParams)
+      return chunkResponse
```

Comment:
I think we should probably change the return type to have something that resembles the `Block` interface from ethers and try to map the values that make sense and leave the others as-is

Reply:
I did some changes, please tell me if that's what you were trying to explained.